app/src/app/components/contents/downloads/FileGroupDownloadButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { useParams } from "react-router-dom";
+import { download, getZipUrl } from "../../../../lib/download";
+import { Button, buttonVariants } from "../../Base/Button";
+import { FolderDown, Loader2 } from "lucide-react";
+import { usePacketOutletContext } from "../../main/PacketOutlet";
+import { FileMetadata } from "../../../../types";
+import { useState } from "react";
+import { filesToSize } from "../../../../helpers";
+import type { VariantProps } from "class-variance-authority";
+
+interface FileGroupDownloadButtonProps {
+  files: FileMetadata[];
+  zipName: string;
+  buttonText?: string;
+  variant?: VariantProps<typeof buttonVariants>["variant"];
+  className?: string;
+}
+
+export const FileGroupDownloadButton = ({
+  files,
+  zipName,
+  buttonText = "Download",
+  variant = "default",
+  className
+}: FileGroupDownloadButtonProps) => {
+  const { packetId } = useParams();
+  const { packet } = usePacketOutletContext();
+  const [error, setError] = useState("");
+  const [downloading, setDownloading] = useState(false);
+
+  const downloadFiles = async () => {
+    if (!packet || !packetId || files.length < 1) {
+      return;
+    }
+    setDownloading(true);
+    await download(getZipUrl(packetId, files), zipName)
+      .then(() => setError(""))
```

Comment:
Maybe error should be reset when you start a download rather than when it successfully completes. If you've had an error then start a new attempt which takes a while, it would be confusing to have the old error from the previous attempt hanging around.